models/stock_export.py

```diff
@@ -0,0 +1,282 @@
+                                                   'Pallet',
+                                                   'Quantity'])
+
+        row = 0
+        for prod, pallets_by_location in pallets_by_prod_by_loc.items():
+            for location, pallets in pallets_by_location.items():
+                for pallet_by_prod in pallets:
+                    row += 1
+                    stock_sheet.write(row, 0, prod.default_code)
+                    stock_sheet.write(row, 1, location.display_name)
+                    stock_sheet.write(row, 2, pallet_by_prod.name)
+                    stock_sheet.write(row, 3, get_prod_qty(prod, pallet_by_prod))
+
+        # 2/2) 'Stock Summary': one entry for each prod
+
+        summary_sheet = _create_sheet('Stock Summary', ['Part Number',
+                                                        'Pallet Count',
+                                                        'Quantity'])
+
+        for row, (prod, prod_pallets) in enumerate(pallets_by_prod.items(), 1):
+            summary_sheet.write(row, 0, prod.default_code)
+            summary_sheet.write(row, 1, len(prod_pallets))
+            summary_sheet.write(row, 2, get_prod_qty(prod, prod_pallets))
+
+        self.__write_workbook(wb, file_name, "Stock File")
+
+    def run_movement_file_export(self):
+        '''
+        Creates movement file, in excel format, summarising the stock
+        received and sent for a given date, i.e. goods-in and goods-out
+        stock moves done, in two different tabs containing:
+            1) Goods In:  Reference, Part number, Pallet, Quantity;
+            2) Goods Out: Reference, Part number, Pallet, Quantity.
+        '''
+        if not self.date:
+            raise ValidationError(_("Date not found."))
+
+        self.field_data = False
+        self.field_name = False
+
+        Move = self.env['stock.move']
+        picking_type_in = self.env.ref('stock.picking_type_in')
+        picking_type_out = self.env.ref('stock.picking_type_out')
+
+        # @todo: check: usin union (no dups) instead of addition
+        picking_types = picking_type_in | picking_type_out
+
+        moves = Move.search([('state', '=', 'done'),
+                             ('picking_type_id', 'in', picking_types.ids),
+                             ('date', '>', self.date + " 00:00:00"),
+                             ('date', '<', self.date + " 23:59:59")])
+
+        in_moves = moves.filtered(lambda m: m.picking_type_id == picking_type_in)
+        out_moves = moves.filtered(lambda m: m.picking_type_id == picking_type_out)
+
+        _logger.info(_('Creating Excel file'))
+        file_name = 'warehouse_movemenet_{TIMESTAMP}.xls'.format(
+            TIMESTAMP=datetime.now())
+        wb = xlwt.Workbook()
+
+        for goods_moves, sheet_name in [(in_moves, "Goods In"),
+                                        (out_moves, "Goods Out")]:
+            sheet = wb.add_sheet(sheet_name)
```

Comment:
The method above used to create a sheet should be used here.

Reply:
Yeah, but the above one is an inner closure, around the `Workbook` instance, so, if you make it an actual method, you should add that to the args. Besides that, another reason why I did not refactor is that I'm no sure of the impact of adding methods to model classes (and btw, is it necessary to add the `@api.model` decorator?).